contrib/platform/src/com/sun/jna/platform/win32/Msi.java

```diff
@@ -218,4 +251,209 @@ public interface Msi extends StdCallLibrary {
+     *   persistent output to a new file or to specify one of the following predefined persistence modes:
+     *   {@link #MSIDBOPEN_CREATEDIRECT} - Create a new database, direct mode read/write.
+     *   {@link #MSIDBOPEN_CREATE} - Create a new database, transact mode read/write.
+     *   {@link #MSIDBOPEN_DIRECT} - Open a database direct read/write without transaction.
+     *   {@link #MSIDBOPEN_READONLY} - Open a database read-only, no persistent changes.
+     *   {@link #MSIDBOPEN_TRANSACT} - Open a database read/write in transaction mode.
+     *   {@link #MSIDBOPEN_PATCHFILE} - Add this flag to indicate a patch file.
+     *
+     * @param phDatabase
+     *   Pointer to the location of the returned database handle.
+     *
+     * @return
+     *  ERROR_CREATE_FAILED - The database could not be created.
+     *  ERROR_INVALID_PARAMETER - One of the parameters was invalid.
+     *  ERROR_OPEN_FAILED - The database could not be opened as requested.
+     *  ERROR_SUCCESS - The function succeeded.
+     */
+    int MsiOpenDatabase(String szDatabasePath, String szPersist, PointerByReference phDatabase);
+
+    /**
+     * The MsiRecordGetString function returns the string value of a record field.
+     *
+     * @param hRecord
+     *   Handle to the record.
+     *
+     * @param iField
+     *   Specifies the field requested.
+     *
+     * @param szValueBuf
+     *   Pointer to the buffer that receives the null terminated string containing the value of the record field. Do not
+     *   attempt to determine the size of the buffer by passing in a null (value=0) for szValueBuf. You can get the size
+     *   of the buffer by passing in an empty string (for example ""). The function then returns ERROR_MORE_DATA and
+     *   pcchValueBuf contains the required buffer size in TCHARs, not including the terminating null character. On
+     *   return of ERROR_SUCCESS, pcchValueBuf contains the number of TCHARs written to the buffer, not including the
+     *   terminating null character.
+     *
+     * @param pcchValueBuf
+     *   Pointer to the variable that specifies the size, in TCHARs, of the buffer pointed to by the variable
+     *   szValueBuf. When the function returns ERROR_SUCCESS, this variable contains the size of the data copied to
+     *   szValueBuf, not including the terminating null character. If szValueBuf is not large enough, the function
+     *   returns ERROR_MORE_DATA and stores the required size, not including the terminating null character, in the
+     *   variable pointed to by pcchValueBuf.
+     *
+     * @return
+     *   ERROR_INVALID_HANDLE - An invalid or inactive handle was supplied.
+     *   ERROR_INVALID_PARAMETER - An invalid parameter was supplied.
+     *   ERROR_MORE_DATA - The provided buffer was too small to hold the entire value.
+     *   ERROR_SUCCESS - The function succeeded.
+     */
+    int MsiRecordGetString(Pointer hRecord, int iField, char[] szValueBuf, IntByReference pcchValueBuf);
```

Comment:
This is not correct if `w32.ascii` is set to `true`. The `char[]` is only correct if `MsiRecordGetStringW` is called. if `w32.ascii` is `true`, `MsiRecordGetStringA` is used. Either this has to be fixed or the whole Msi library must be bound with `W32APIOptions.UNICODE_OPTIONS`.
This applies to all methods with similar construct.